### ISSUE TITLE
Using system DPI setting on Linux ?

### ISSUE THREAD

**Issue by jcelerier** [processing/processing#5758]
Hi, 

@benfry you mentioned in processing/processing#4183

> There's no API for this on Linux, so you have to set it in Preferences.

but there is a standard way to set the DPI on X11 which is Xft.dpi in X resources:

 (giving you the C code because I don't know how to access the X11 API through Java but still...)

```C
#include <X11/Xlib.h>
#include <X11/Xresource.h>
#include <cstdio>

int main (int argc, char *argv[])
{
    Display* display = XOpenDisplay (0);
    XrmInitialize();
    XrmDatabase xrdb = XrmGetStringDatabase(XResourceManagerString(display));
    
    XrmValue v;
    char* ret_str;
    int ok = XrmGetResource(xrdb, "Xft.dpi", "String", &ret_str, &v);
    printf("DPI is %s\n", v.addr);    
}
```

**Comment by benfry** [comment]
Yeah, thanks… C code is problematic but there are other options [discussed here](https://github.com/processing/processing/issues/4895) that might work. None of them ideal; perhaps Java 11 finally has a more standard API for it, though we're also at the mercy of the window manager.
